### PR TITLE
Add ability to set rubocop version

### PR DESCRIPTION
**Pull request by AlexeyMatskevich**
Solution for #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,30 @@ It's same as `-level` flag of reviewdog.
 Optional. Reporter of reviewdog command [`github-pr-check`, `github-pr-review`].
 The default is `github-pr-check`.
 
+### `rubocop_version`
+
+**Required**. Set rubocop version.
```
Comment (mgrachev) [inline README.md:52]: I would like these flags to be optional. I mean if a user doesn't specify the version, we will install the latest version

**Comment by mgrachev** [comment]
@AlexeyMatskevich 👋Thank you for your contribution!

I have left a comment. Could you please fix it?

File: action.yml [hunk 2]
```diff
@@ -19,6 +19,24 @@ inputs:
       Reporter of reviewdog command [github-pr-check,github-pr-review].
       Default is github-pr-check.
     default: 'github-pr-check'
+  rubocop_version:
+    description: 'Rubocop version'
+    required: false
```
Comment (mgrachev) [inline action.yml:24]: As I can see, other inputs don't use `required: false`. Can we omit it?

File: README.md [hunk 3]
```diff
@@ -47,6 +47,30 @@ It's same as `-level` flag of reviewdog.
 Optional. Reporter of reviewdog command [`github-pr-check`, `github-pr-review`].
 The default is `github-pr-check`.
 
+### `rubocop_version`
+
+Set rubocop version.
```
Comment (mgrachev) [inline README.md:52]:
```suggestion
Optional. Set rubocop version. 
```

File: README.md [hunk 4]
```diff
@@ -47,6 +47,30 @@ It's same as `-level` flag of reviewdog.
 Optional. Reporter of reviewdog command [`github-pr-check`, `github-pr-review`].
 The default is `github-pr-check`.
 
+### `rubocop_version`
+
+Set rubocop version. 
+
+### `rubocop_rails_version`
+
+Set rubocop rails version.
```
Comment (mgrachev) [inline README.md:56]:
```suggestion
Optional. Set rubocop rails version.
```

File: README.md [hunk 5]
```diff
@@ -47,6 +47,30 @@ It's same as `-level` flag of reviewdog.
...
+
+### `rubocop_performance_version`
+
+Set rubocop performance version.
```
Comment (mgrachev) [inline README.md:60]:
```suggestion
Optional. Set rubocop performance version.
```

File: README.md [hunk 6]
```diff
@@ -47,6 +47,30 @@ It's same as `-level` flag of reviewdog.
...
+
+### `rubocop_rspec_version`
+
+Set rubocop rspec version.
```
Comment (mgrachev) [inline README.md:64]:
```suggestion
Optional. Set rubocop rspec version.
```

File: README.md [hunk 7]
```diff
@@ -47,6 +47,30 @@ It's same as `-level` flag of reviewdog.
...
+
+### `rubocop_i18n_version`
+
+Set rubocop i18n version.
```
Comment (mgrachev) [inline README.md:68]:
```suggestion
Optional. Set rubocop i18n version.
```

File: README.md [hunk 8]
```diff
@@ -47,6 +47,30 @@ It's same as `-level` flag of reviewdog.
...
+
+### `rubocop_rake_version`
+
+Set rubocop rake version.
```
Comment (mgrachev) [inline README.md:72]:
```suggestion
Optional. Set rubocop rake version.
```

**Comment by mgrachev** [comment]
@AlexeyMatskevich What do you think about giving users the ability to specify rubocop extensions by manual?

For example, we will have the ability to specify the `rubocop` gem's version via the `rubocop_version` input and at the same time we will have the ability to specify rubocop extensions via the `rubocop_extensions` input like this: 
```
rubocop_extensions: rubocop-performance:1.5.1 rubocop-minitest
```

**Comment by AlexeyMatskevich** [comment]
I think I can try to do this using Shell Script and parse the input to the generate corresponding gems installation command

File: README.md [hunk 9]
```diff
@@ -47,6 +47,17 @@ It's same as `-level` flag of reviewdog.
...
+### `rubocop_version`
+
+Optional. Set rubocop version. 
+By default install latest version.
+
+### `rubocop_extensions`
+
+Optional. Set list of rubocop extensions with versions. 
+
+By default install rubocop-rails rubocop-performance rubocop-rspec rubocop-i18n rubocop-rake with latest versions.
```
Comment (mgrachev) [inline README.md:59]:
```suggestion
By default install `rubocop-rails`, `rubocop-performance`, `rubocop-rspec`, `rubocop-i18n`, `rubocop-rake` with latest versions.
```

File: README.md [hunk 10]
```diff
@@ -47,6 +47,17 @@ It's same as `-level` flag of reviewdog.
 Optional. Reporter of reviewdog command [`github-pr-check`, `github-pr-review`].
 The default is `github-pr-check`.
 
+### `rubocop_version`
```
Comment (mgrachev) [inline README.md:50]: This one should go after `rubocop_flags `.

File: action.yml [hunk 11]
```diff
@@ -19,6 +19,10 @@ inputs:
       Reporter of reviewdog command [github-pr-check,github-pr-review].
       Default is github-pr-check.
     default: 'github-pr-check'
+  rubocop_version:
+    description: 'Rubocop version'
+  rubocop_extensions:
+    description: 'Rubocop extensions'
```
Comment (mgrachev) [inline action.yml:25]:
```suggestion
    description: 'Rubocop extensions'
    default: 'rubocop-rails rubocop-performance rubocop-rspec rubocop-i18n rubocop-rake'
```
Please use the `default` option to keep the list of extensions which install by default.

File: action.yml [hunk 12]
```diff
@@ -28,6 +32,12 @@ runs:
     - ${{ inputs.tool_name }}
     - ${{ inputs.level }}
     - ${{ inputs.reporter }}
+    - ${{ inputs.rubocop_version }}
+    - ${{ inputs.rubocop_rails_version }}
+    - ${{ inputs.rubocop_performance_version }}
+    - ${{ inputs.rubocop_rspec_version }}
+    - ${{ inputs.rubocop_i18n_version }}
+    - ${{ inputs.rubocop_rake_version }}
```
Comment (mgrachev) [inline action.yml:40]: Are these old changes?

File: action.yml [hunk 13]
```diff
@@ -28,6 +32,12 @@ runs:
     - ${{ inputs.tool_name }}
     - ${{ inputs.level }}
     - ${{ inputs.reporter }}
+    - ${{ inputs.rubocop_version }}
+    - ${{ inputs.rubocop_rails_version }}
```
Comment (mgrachev) [inline action.yml:36]:
```suggestion
    - ${{ inputs.rubocop_extensions }}
```

File: entrypoint.sh [hunk 14]
```diff
@@ -1,8 +1,25 @@
...
+  else
+    gem install -N rubocop-rails rubocop-performance rubocop-rspec rubocop-i18n rubocop-rake
+  fi
+}
 
 cd "$GITHUB_WORKSPACE"
 
 export REVIEWDOG_GITHUB_API_TOKEN="${INPUT_GITHUB_TOKEN}"
 
+gem install rubocop $(version $INPUT_RUBOCOP_VERSION)
```
Comment (mgrachev) [inline entrypoint.sh:20]:
```suggestion
gem install -N rubocop $(version $INPUT_RUBOCOP_VERSION)
```

File: entrypoint.sh [hunk 15]
```diff
@@ -1,8 +1,25 @@
 #!/bin/sh
+version() {
+  if [ -n "$1" ]; then
+    echo "-v $1"
+  fi
+}
+
+install_extensions() {
+  if [ -n "$1" ]; then
+    echo "$1" | sed "s/[^ ][^ ]*/'&'/g" | xargs gem install
```
Comment (mgrachev) [inline entrypoint.sh:10]: Can we install extensions via the command: `gem install -N $INPUT_RUBOCOP_VERSION`?

Comment (AlexeyMatskevich) [reply entrypoint.sh:10]: No we cannot specify version of gem with -N flaf

File: entrypoint.sh (hunk 14 above)
Comment (AlexeyMatskevich) [reply entrypoint.sh:20]: We cannot cpecify gem version with -N flag

File: action.yml (hunk 12 above)
Comment (AlexeyMatskevich) [reply action.yml:40]: Yep

File: entrypoint.sh (hunk 14 above)
Comment (mgrachev) [reply entrypoint.sh:20]: The `-N` argument is used to disable documentation generation.
https://guides.rubygems.org/command-reference/#gem-install

File: entrypoint.sh (hunk 15 above)
Comment (mgrachev) [reply entrypoint.sh:10]: Sorry, I mean the command to install extensions: `gem install -N $INPUT_RUBOCOP_EXTENSIONS `

File: entrypoint.sh (hunk 14 above)
Comment (AlexeyMatskevich) [reply entrypoint.sh:20]: Sorry here, I misunderstood everything and everything works with the flag -N

File: entrypoint.sh (hunk 15 above)
Comment (AlexeyMatskevich) [reply entrypoint.sh:10]: I get this result
```sh
INPUT_RUBOCOP_EXTENSIONS='rubocop-performance:1.5.1 rubocop-minitest'
gem install -N $INPUT_RUBOCOP_EXTENSIONS
ERROR:  Could not find a valid gem 'rubocop-performance:1.5.1 rubocop-minitest' (>= 0) in any repository
```
but I guessed that everything is installed with a simple parameter passing through xargs
```sh
echo $INPUT_RUBOCOP_EXTENSIONS | xargs gem install -N                
Successfully installed rubocop-performance-1.5.1
Fetching rubocop-minitest-0.9.0.gem
Successfully installed rubocop-minitest-0.9.0
2 gems installed
```

**Comment by mgrachev** [comment]
@AlexeyMatskevich 🙏Thanks a lot! You have done a wonderful job! ❤️

**Pull request merged**